### DISCUSSION TITLE
Squash Add-On doesn't squash Weblate changes when set to 'by Author'

### DISCUSSION THREAD

**Discussion by Mark-Giesen** [WeblateOrg/weblate#10091] [Q&A]
I configured the Squash Add-On to squash the commits by author. This seems to work fine, except for the commits by Weblate itself. When Weblate changes something, like the header of a .po file, I expect this to be squashed all together because the author is Weblate. Is this an error/oversight or intentional. Can I change this somehow?

**Comment by nijel** [comment]
It should squash everything. What can happen is that another add-on generates commit after squashing. This can happen as there is no explicit ordering of add-on execution.

**Reply by Mark-Giesen** [reply]
Ah, so after squach add-on has run, the prefill translation might run and produce a lot of commits when a new string is added. Do I understand right that If I now press push in maintenance, all these commits of the prefill add-on will be squached into one?

**Reply by Mark-Giesen** [reply]
I tried this last week on a PR. It had some commits of actual translators and one per language created by Weblate because of an added string. I pressed commit in Weblate maintenance followed by push. I expected the squash add-on to run and squash all Weblate commits into one, but this didn't happen. So I must misunderstand something.

**Reply by nijel** [reply]
Can you share link to a Weblate project and/or the pull request so that I can take a look?

**Reply by Mark-Giesen** [reply]
It's [this project](https://hosted.weblate.org/projects/hedy/). The PR is merged in the meantime, after a lot of (not weblate related) merge conflict handling and tests. Not much to be seen anymore. I'll see if I can recreate a set as mentioned above and leave it for you to see. Thanks for willing to take a look, I'll be back.

**Reply by Mark-Giesen** [reply]
I just added a new string to the source language file: translations/en/LC_MESSAGES/messages.po and Weblate created copies of this in all languages. This time I didn't see every file as a separate commit on [this PR](https://github.com/hedyorg/hedy/pull/4609). I see commits bundled by author and one of them is weblate. Title is little weird: Translated using Weblate (Kurdish (Northern)). Probably the first or last file weblate encountered. So this is what's expected and probably the squash add-on ran after the fill translations add-on this time.

**Reply by Mark-Giesen** [reply]
Not sure if more help is needed for now, I'll come back here if I see something weird that I need addressed after all.
